lib/HeroBonus.cpp

```diff
@@ -200,13 +200,25 @@ int CTotalsProxy::getValue() const
 		value = initialValue + bonuses->totalValue();
 		cachedLast = treeVersion;
 	}
+	return value;
+}
+
+int CTotalsProxy::getValueAndList(TConstBonusListPtr & bonusList) const
+{
+	const auto treeVersion = target->getTreeVersion();
+	bonusList = target->getBonuses(selector);

```

Comment:
We could cache bonuses list too, and may be even make `CTotalsProxy` a descendant of `CBonusProxy` (but do not merge completely -  `totalValue()` is somewhat costly operation)

Reply:
Done.  lib/HeroBonus.h  and lib/HeroBonus.cpp was changed.
But may be it's excessive because `BonusList`  caching is already present in the `CBonusSystemNode::getAllBonuses`

Reply:
This class was made because cache in `CBonusSystemNode` is too slow.